src/game/scenes/lobby.c

```diff
@@ -1491,6 +1548,7 @@ void lobby_tick(scene *scene, int paused) {
 
                                 // force the speed to 3
                                 game_state_set_speed(gs, 10);
+                                game_state_copy_match_settings(gs, &local->opponent->match_settings);
```

Comment:
We should do this differently, by having the lobby pack the challengee's match settings into the spectate packet stream, I think.